spec/std/process_spec.cr

```diff
@@ -85,6 +85,79 @@ describe Process do
     $?.exit_code.should eq(0)
   end
 
+  it "getuid" do
+    Process.effective_user_id.should eq `id -u`.to_i
+    Process.effective_user_id.should eq Process.effective_user_id
+  end
+
+  it "getgid" do
+    Process.group_id.should eq `id -g`.to_i
+    Process.group_id.should eq Process.effective_group_id
+  end
+
+  it "become_user" do
+    success = false
+    ruid = Process.user_id
+    euid = Process.effective_user_id
+
+    begin
+      # Leave ruid at 0 to switch back
+      Process.become_user(ruid: 0, euid: 8888, suid: 9999)
+      Process.user_id.should eq 0
+      Process.effective_user_id.should eq 8888
+      success = true
+    rescue ex : Errno
+      # Ok for non-root
+    ensure
+      Process.become_user(ruid: ruid, euid: euid) if success
+    end
+
+    if euid == 0
```

Comment:
This spec is hard to follow. Why not streamline flow like this?

```cr
if euid == 0
    Process.become_user(ruid: 0, euid: 8888, suid: 9999)
    Process.user_id.should eq 0
    Process.effective_user_id.should eq 8888
    Process.become_user(ruid: ruid, euid: euid)
else
  expect_raises_errno(Errno::EPERM) do
    Process.become_user(ruid: 0, euid: 8888, suid: 9999)
  end
end
```

Or maybe even better, separate both cases into individual examples and only execute them according to the value of `euid`. This clearly communicates what is actually being tested.
```cr
if euid == 0
  it ".become_user (root)" do
    # ...
  end
else
  it ".become_user (non-root)" do
    # ....
  end
end
```